packages/react-aria-components/src/HiddenDateInput.tsx

```diff
@@ -0,0 +1,132 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+
+import {CalendarDate, CalendarDateTime, parseDate, parseDateTime} from '@internationalized/date';
+import {DateFieldState, DatePickerState, DateSegmentType} from 'react-stately';
+import React, {ReactNode} from 'react';
+import {useVisuallyHidden} from 'react-aria';
+
+interface AriaHiddenDateInputProps {
+  /**
+   * Describes the type of autocomplete functionality the input should provide if any. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefautocomplete).
+   */
+  autoComplete?: string,
+  /** HTML form input name. */
+  name?: string,
+  /** Sets the disabled state of the input. */
+  isDisabled?: boolean
+}
+
+interface HiddenDateInputProps extends AriaHiddenDateInputProps {
+  /**
+   * State for the input.
+   */
+  state: DateFieldState | DatePickerState
+}
+
+export interface HiddenDateAria {
+  /** Props for the container element. */
+  containerProps: React.HTMLAttributes<HTMLDivElement>,
+  /** Props for the hidden input element. */
+  inputProps: React.InputHTMLAttributes<HTMLInputElement>
+}
+
+const dateSegments = ['day', 'month', 'year'];
+const granularityMap = {'hour': 1, 'minute': 2, 'second': 3};
+
+export function useHiddenDateInput(props: HiddenDateInputProps, state: DateFieldState | DatePickerState) : HiddenDateAria {
+  let {
+    autoComplete,
+    isDisabled,
+    name
+  } = props;
+  let {visuallyHiddenProps} = useVisuallyHidden();
+
+  let inputStep = 60;
+  if (state.granularity === 'second') {
+    inputStep = 1;
+  } else if (state.granularity === 'hour') {
+    inputStep = 3600; 
+  }
+
+  let dateValue = state.value == null ? '' : state.value.toString();
+
+  let inputType = state.granularity === 'day' ? 'date' : 'datetime-local';
+
+  let timeSegments = ['hour', 'minute', 'second'];
+  // Depending on the granularity, we only want to validate certain time segments
+  let end = 0;
+  if (timeSegments.includes(state.granularity)) {
+    end = granularityMap[state.granularity];
+    timeSegments = timeSegments.slice(0, end);
+  }
+
+  return {
+    containerProps: {
+      ...visuallyHiddenProps,
+      'aria-hidden': true,
+      // @ts-ignore
+      ['data-react-aria-prevent-focus']: true,
+      // @ts-ignore
+      ['data-a11y-ignore']: 'aria-hidden-focus'
+    },
+    inputProps: {
+      tabIndex: -1,
+      autoComplete,
+      disabled: isDisabled,
+      type: inputType,
+      // We set the form prop to an empty string to prevent the hidden date input's value from being submitted
+      form: '',
+      name,
+      step: inputStep,
+      value: dateValue,
+      onChange: (e) => {
+        let targetString = e.target.value.toString();
+        if (targetString) {
+          let targetValue: CalendarDateTime | CalendarDate = parseDateTime(targetString);
```

Comment:
We should probably add a try...catch around these parsing functions in case the value is somehow invalid.